x-pack/plugins/apm/server/lib/index_pattern/index.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Server } from 'hapi';
+// @ts-ignore
+import { getSavedObjects } from '../../../../../../src/legacy/core_plugins/kibana/server/tutorials/apm/saved_objects/get_saved_objects';
+import { getSavedObjectsClient } from '../saved_objects/client';
+
+export async function ensureIndexPatternExists(server: Server) {
+  const config = server.config();
+  const apmIndexPatternTitle = config.get('apm_oss.indexPattern');
+  const savedObjectsClient = getSavedObjectsClient(server);
+  const apmIndexPatternSearchResp = await savedObjectsClient.find({
+    type: 'index-pattern',
+    search: `"${apmIndexPatternTitle}"`,
+    searchFields: ['title'],
+    perPage: 200
+  });
```

Comment:
I seem to recall that the saved objects API uses some fuzzy match and not exact match. So the search query might return results, even if the exact APM index doesn't exist. 
Right now it looks like you expect it to match exactly (`apmIndexPatternSearchResp.total === 0`) - please double check this.

if we expect it to be an exact match, why `perPage: 200`? Shouldn't it then be `perPage: 1`?